From bd07e6f6a5e7e4fe4b0739ee3736451f56932ba1 Mon Sep 17 00:00:00 2001
From: Jakub Sosnovec <jakub.sosnovec@nxp.com>
Date: Tue, 22 Sep 2020 11:02:59 +0200
Subject: [PATCH 08/17] pcie: dma: Set send_signal_to_user and fix memory
 corruption for siginfo

commit bd07e6f6a5e7e4fe4b0739ee3736451f56932ba1 from
https://source.codeaurora.org/external/autobsps32/linux

This is a combination of the following commits:
    Set send_signal_to_user after DMA finishes [6dd2c01d8e02]
    Fix memory corruption for siginfo [5903b976707f]

Upstream-Status: Pending

Issue: ALB-5982
Signed-off-by: Jakub Sosnovec <jakub.sosnovec@nxp.com>
Signed-off-by: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/pci/controller/dwc/pci-s32v234.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/controller/dwc/pci-s32v234.c b/drivers/pci/controller/dwc/pci-s32v234.c
index e1a0c6fdc118..71c757bf6712 100644
--- a/drivers/pci/controller/dwc/pci-s32v234.c
+++ b/drivers/pci/controller/dwc/pci-s32v234.c
@@ -1317,6 +1317,8 @@ static int s32v234_pcie_probe(struct platform_device *pdev)
 	pcie->dev = dev;
 	pcie->ops = &s32v234_pcie_ops;
 
+	s32v234_pp->send_signal_to_user = send_signal_to_user;
+
 	/* Added for PCI abort handling */
 	dbi_base = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	pcie->dbi_base = devm_ioremap_resource(dev, dbi_base);
@@ -1389,7 +1391,7 @@ static int s32v234_pcie_probe(struct platform_device *pdev)
 		}
 		dw_pcie_dma_clear_regs(pcie, &s32v234_pp->dma);
 
-		memset(&s32v234_pp->info, 0, sizeof(struct siginfo));
+		memset(&s32v234_pp->info, 0, sizeof(struct kernel_siginfo));
 		s32v234_pp->info.si_signo = SIGUSR1;
 		s32v234_pp->info.si_code = SI_USER;
 		s32v234_pp->info.si_int = 0;
-- 
2.17.1

